src/app/CASESessionManager.cpp

```diff
@@ -69,6 +69,50 @@ void CASESessionManager::FindOrEstablishSession(const ScopedNodeId & peerId, Cal
     session->Connect(onConnection, onFailure);
 }
 
+void CASESessionManager::FindOrEstablishSession(const ScopedNodeId & peerId, Callback::Callback<OnDeviceConnected> * onConnection,
+                                                Callback::Callback<OperationalSessionSetup::OnSetupFailure> * onSetupFailure
+#if CHIP_DEVICE_CONFIG_ENABLE_AUTOMATIC_CASE_RETRIES
+                                                ,
+                                                uint8_t attemptCount, Callback::Callback<OnDeviceConnectionRetry> * onRetry
+#endif
+)
+{
+    ChipLogDetail(CASESessionManager, "FindOrEstablishSession: PeerId = [%d:" ChipLogFormatX64 "]", peerId.GetFabricIndex(),
```

Comment:
Can we please share the code here with the onFailure case by having a helper function that takes both callbacks, one of which is null?  It would be really good not to duplicate the general logic here....